types/src/transaction/mod.rs

```diff
@@ -529,15 +674,153 @@ pub enum TransactionPayload {
     /// A multisig transaction that allows an owner of a multisig account to execute a pre-approved
     /// transaction as the multisig account.
     Multisig(Multisig),
+    /// A new transaction payload format with support for versioning.
+    /// Contains an executable (script/entry function) along with extra configuration.
+    /// Once this new format is fully rolled out, above payload variants will be deprecated.
+    Payload(TransactionPayloadInner),
+}
+
+#[derive(Clone, Debug, Hash, Eq, PartialEq, Serialize, Deserialize)]
+pub enum TransactionPayloadInner {
+    V1 {
+        executable: TransactionExecutable,
+        extra_config: TransactionExtraConfig,
+    },
+}
+
+#[derive(Clone, Debug, Hash, Eq, PartialEq, Serialize, Deserialize)]
+pub enum TransactionExecutable {
+    Script(Script),
+    EntryFunction(EntryFunction),
+    Empty,
```

Comment:
what is empty used for?

Reply:
Currently, a multisig transaction can have an empty payload as in here: https://github.com/aptos-labs/aptos-core/blob/main/types/src/transaction/multisig.rs#L15.

To accommodate this case, I added empty executable here. In Aptos VM, I have checks to ensure that empty executable is accepted only when multisig_address is set. For non-multisig transactions, if it is empty executable, we discard the transaction.